### DISCUSSION TITLE
local-docker-compose 파일을 만드신 이유가 궁금합니다!

### DISCUSSION THREAD

**Discussion by kssumin** [JNU-econovation/EEOS-BE#157] [Q&A]
## 개발용, 운영용 docker-compose 파일이 존재한 이유
---
* 개발용 서버, 운영용 서버에서 docker-compose 명령어 하나로 연관되어 있는 DB와 WAS 서버를 한꺼번에 띄우기 위함이였습니다.

## 개발용, 운영용 파일을 분리한 이유는?
---
### 1. 환경별 요구사항이 다르기 때문
* 운영용 서버에서는 개발용 서버와는 달리 Nginx 인프라가 추가적으로 필요합니다.
* 개발 환경: 백엔드 서비스에 직접 접근 필요 (8080 포트 노출)
* 운영 환경: Nginx를 통한 간접 접근 (50001 포트를 통해 Nginx에 접근, 백엔드는 내부 네트워크에만 노출)

### 2. 설정의 명확성:
각 환경별로 파일을 분리함으로써, 해당 환경에 필요한 설정만 명확하게 볼 수 있습니다.

## 로컬용 docker-compose가 필요한 이유가 궁금합니다!!
---
저는 local용 docker-compose가 필요하지 않을 것이라고 생각한 이유는 resources/docker-compose.yml 파일을 통해 WAS 서버를 제외한 전체 개발 환경을 구성할 수 있기 떄문인데요.
(+ WAS서버는 intelliJ를 통해서 실행을 할 것이라고 생각)

제가 생각하지 못 한 다른 이유가 있을 것 같아..!(또 해당 파일을 만드신 분의 어떤 이슈를 겪으셔서 해당 파일을 작업하시게 되셧는지!)가 궁금합니다!

**Comment by rlajm1203** [comment]
음 생각해보니 local용을 따로 만들었을 때 어떤 큰 이슈가 있었던 것은 아닙니다!

정확히 기억은 안 나지만, 로컬에서 코드를 작성하고 테스트를 하고 싶었는데, IntelliJ로 실행을 못할 거라고 생각했어서 local용을 따로 만든 것 같네요..!
(MySQL, Redis를 로컬로 항상 실행하고 있어야 하기 떄문에..)
